spec/lib/amazon_ses_mailer/message_spec.rb

```diff
@@ -0,0 +1,174 @@
+                                                                                     content:                 {
+                                                                                       template: {
+                                                                                         template_name: options[:template],
+                                                                                         template_data: options[:merge_vars]
+                                                                                       }
+                                                                                     },
+                                                                                     configuration_set_name:  options[:configuration_set_name],
+                                                                                     list_management_options: {
+                                                                                       contact_list_name: options_with_contact_list_name[:contact_list_name],
+                                                                                       topic_name:        options_with_contact_list_name[:topic_name]
+                                                                                     } })
+      end
+    end
+  end
+
+  context '#ensure_array' do
+    context 'when array_or_string is array' do
+      let(:array_or_string) { [Faker::Internet.safe_email] }
+
+      it 'should should return array of string as is' do
+        expect(message.send(:ensure_array, array_or_string)).to eq(array_or_string)
+      end
+    end
+
+    context 'when array_or_string is string' do
+      let(:array_or_string) { Faker::Internet.safe_email }
+
+      it 'should should return array of array contains string' do
+        expect(message.send(:ensure_array, array_or_string)).to eq([array_or_string])
+      end
+    end
+
+    context 'when array_or_string not array or string' do
+      let(:array_or_string) { nil }
+
+      it 'should should return empty array' do
+        expect(message.send(:ensure_array, array_or_string)).to be_empty
+      end
+    end
+  end
+
+  context '#build_list_management_options' do
+    context 'when options contains contact_list_name present' do
+      let(:options) { { contact_list_name: 'test', topic_name: 'email' } }
+
+      it 'should return contact_list_name hash as { contact_list_name, topic_name }' do
+        expect(message.send(:build_list_management_options, options)).to eq({ contact_list_name: options[:contact_list_name], topic_name: options[:topic_name] })
+      end
+    end
+
+    context 'when options contains contact_list_name is not set' do
+      let(:options) { { topic_name: 'email' } }
+
+      it 'should return contact_list_name hash as { contact_list_name, topic_name }' do
+        expect(message.send(:build_list_management_options, options)).to be_nil
+      end
+    end
+  end
+
+  context '#delivering?' do
```

Comment:
Add tests where 2 or more interceptors are registered, and it should be truthy only when all of them allow delivery, and false otherwise. This is basically testing the `all?` method.